libs/SalesforceSDK/src/com/salesforce/androidsdk/analytics/AnalyticsPublisherService.kt

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (c) 2016-present, salesforce.com, inc.
+ * All rights reserved.
+ * Redistribution and use of this software in source and binary forms, with or
+ * without modification, are permitted provided that the following conditions
+ * are met:
+ * - Redistributions of source code must retain the above copyright notice, this
+ * list of conditions and the following disclaimer.
+ * - Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ * - Neither the name of salesforce.com, inc. nor the names of its contributors
+ * may be used to endorse or promote products derived from this software without
+ * specific prior written permission of salesforce.com, inc.
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.salesforce.androidsdk.analytics
+
+import android.content.Context
+import androidx.work.ListenableWorker.Result.failure
+import androidx.work.ListenableWorker.Result.success
+import androidx.work.OneTimeWorkRequest.Companion.from
+import androidx.work.WorkManager.getInstance
+import androidx.work.Worker
+import androidx.work.WorkerParameters
+import com.salesforce.androidsdk.accounts.UserAccountManager
+
+/**
+ * An Android background tasks worker which publishes stored analytics.
+ * This class is intended to be instantiated by the background tasks work
+ * manager.
+ *
+ * Use startActionPublish to enqueue an analytics publishing worker.
+ *
+ * @param context The Android context provided by the work manager
+ * @param workerParams The worker parameters provided by the work manager
+ * @see <a href='https://developer.android.com/guide/background'>Android Background Tasks</a>
+ */
+internal class AnalyticsPublishingWorker(
```

Comment:
@brandonpage, I'd be curious to get your early feedback on this draft Background Tasks `Worker` for analytics. This is the easier of the two objects have to migrate, so I started here.  Is this heading in the direction you imagined?

The app will run and fire this worker, even in this rough draft state.  I'm looking now for any enqueue options that we might want to consider beyond the defaults.

Reply:
Sorry for the late response.  I took a look at the updated code and it looks very clean.  👍